### PR TITLE
Fixes Intellicard view-locking bug

### LINKED ISSUE #20653: AI carding doesn't work
when you're carded your view is locked to wherever you were looking when carded, not where you actually are.

also happens if you are transferred into a mech.

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
hopefully fixes #20653 

newbie coder feeling S T R O N G because they can add an extra proc.

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -639,6 +639,7 @@
 			AI.control_disabled = 1
 			AI.radio_enabled = 0
 			AI.loc = card
```
Comment (optimumtact) [inline code/game/mecha/mecha.dm]: you don't need to both update the loc and use forceMove

force move does it and handles other things like camera view, in general it is preferred over simply setting the loc for that reason.

**Review by optimumtact — CHANGES_REQUESTED** [review]

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [inline code/game/mecha/mecha.dm]: so this portion isn't even an issue?

Comment (optimumtact) [inline code/game/mecha/mecha.dm]: no it is an issue, you just need to *replace* the code that does AI.loc = card
with the forceMove

Same goes with the rest of the instances

Comment (ExcessiveUseOfCobblestone) [inline code/game/mecha/mecha.dm]: ohhhhhhhhhhhhhh

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -761,6 +761,7 @@ var/list/ai_list = list()
 		radio_enabled = 0 	//No talking on the built-in radio for you either!
 		loc = card//Throw AI into the card.
 		card.AI = src
+		user.forceMove(src)
```
Comment (optimumtact) [inline code/modules/mob/living/silicon/ai/ai.dm]: replace the line two above

    loc = card

with 

   forceMove(card) //basically the same as src.forceMove(card), also don't include this comment

**Review by optimumtact — APPROVED** [review]
LGTM

File: code/game/objects/structures/ai_core.dm [hunk 3]
```diff
@@ -240,7 +240,7 @@ atom/proc/transfer_ai(interaction, mob/user, mob/living/silicon/ai/AI, obj/item/
 	if(interaction == AI_TRANS_FROM_CARD)
 		AI.control_disabled = 0
 		AI.radio_enabled = 1
-		AI.loc = loc//To replace the terminal.
+		AI.forceMove(card) // to replace the terminal.
```
Comment (phil235) [inline code/game/objects/structures/ai_core.dm]: AI.forceMove(loc)

Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/structures/ai_core.dm]: check now

**Pull request merged**